nilearn/tests/test_niimg.py

```diff
@@ -49,8 +49,11 @@ def test_get_target_dtype():
     dtype_kind_float = niimg._get_target_dtype(get_data(img).dtype,
                                                target_dtype='auto')
     assert dtype_kind_float == np.float32
-
-    img2 = Nifti1Image(np.ones((2, 2, 2), dtype=np.int64), affine=np.eye(4))
+    # Passing dtype or header is required when using int64
+    # https://nipy.org/nibabel/changelog.html#api-changes-and-deprecations
+    hdr = Nifti1Header()
```

Comment:
Do you mean for these to be eventually output as float32?

```Python
>>> import nibabel as nb
>>> nb.Nifti1Header().get_data_dtype()
dtype('<f4')
```

If you want to record the intended dtype, you need to do `hdr.set_data_dtype()`.

Reply:
For this sort of test, it might be useful to round-trip images to see what happens when serialized. (You might have a helper function already...)

```Python
def rt_img(img):
    return img.from_bytes(img.to_bytes())
```

Reply:
Hi @effigies thanks for pointing that out. You are right I should add a line with `hdr.set_data_dtype(np.int64)`. 
I can also add the check with the round trip function. I'm not sure if we actually have such a function but might be useful. Also in this case would it be enough to replace L57 below:

`assert get_data(img2).dtype.kind == 'i'`

with

`assert img2.get_data_dtype().kind == 'i'`